test/spark/run-gc-test.sh

```diff
@@ -15,11 +19,35 @@ run_lakectl() {
+run_gc_according_to_storage_provider() {
+  local repo=$1
+  local additional_config_string=$2
+  if [[ ${LAKEFS_BLOCKSTORE_TYPE} == "azure" ]]; then
+    docker-compose run -v ${CLIENT_JAR}:/client/client.jar -T --no-deps --rm spark-submit bash -c "spark-submit -v --packages org.apache.hadoop:hadoop-azure:3.2.1 --master spark://spark:7077 --class io.treeverse.clients.GarbageCollector ${additional_config_string} -c spark.hadoop.lakefs.api.url=http://docker.lakefs.io:8000/api/v1 -c spark.hadoop.lakefs.api.access_key=\${TESTER_ACCESS_KEY_ID} -c spark.hadoop.lakefs.api.secret_key=\${TESTER_SECRET_ACCESS_KEY} -c spark.hadoop.lakefs.api.connection.timeout_seconds=3 -c spark.hadoop.lakefs.api.read.timeout_seconds=8 -c spark.hadoop.fs.azure.account.key.\${LAKEFS_BLOCKSTORE_AZURE_STORAGE_ACCOUNT}.dfs.core.windows.net=\${LAKEFS_BLOCKSTORE_AZURE_STORAGE_ACCESS_KEY} ${parallelism} /client/client.jar ${repo}"
   else
-    docker-compose run -v ${CLIENT_JAR}:/client/client.jar -T --no-deps --rm spark-submit bash -c "spark-submit -v --master spark://spark:7077 --class io.treeverse.clients.GarbageCollector -c spark.hadoop.lakefs.api.url=http://docker.lakefs.io:8000/api/v1 -c spark.hadoop.lakefs.api.access_key=\${TESTER_ACCESS_KEY_ID} -c spark.hadoop.lakefs.api.secret_key=\${TESTER_SECRET_ACCESS_KEY} -c spark.hadoop.fs.s3a.access.key=\${AWS_ACCESS_KEY_ID} -c spark.hadoop.fs.s3a.secret.key=\${AWS_SECRET_ACCESS_KEY} $parallelism /client/client.jar $1 us-east-1"
+    docker-compose run -v ${CLIENT_JAR}:/client/client.jar -T --no-deps --rm spark-submit bash -c "spark-submit -v --master spark://spark:7077 --class io.treeverse.clients.GarbageCollector -c spark.hadoop.lakefs.api.url=http://docker.lakefs.io:8000/api/v1 ${additional_config_string} -c spark.hadoop.lakefs.api.access_key=\${TESTER_ACCESS_KEY_ID} -c spark.hadoop.lakefs.api.secret_key=\${TESTER_SECRET_ACCESS_KEY} -c spark.hadoop.fs.s3a.access.key=\${AWS_ACCESS_KEY_ID} -c spark.hadoop.fs.s3a.secret.key=\${AWS_SECRET_ACCESS_KEY} ${parallelism} /client/client.jar ${repo} us-east-1"
+  fi
+}
+
+run_gc () {
+  local repo=$1
+  local run_mode=$2
+  local mark_id=$3
+  local sweep_config=""
+  local mark_id_config=""
+  echo "mark id = ${mark_id}, gc_mode = ${run_mode}"
+  if [ ${run_mode} -eq ${MARK_ONLY_MODE} ] || [ ${run_mode} -eq ${SWEEP_ONLY_MODE} ]; then
+    # If we are in sweep-only mode, we'll first mark and afterwards we'll sweep
+    sweep_config="-c spark.hadoop.lakefs.gc.do_sweep=false "
+  fi
+  if [[ ${mark_id} != "" ]]; then
+    mark_id_config="-c spark.hadoop.lakefs.gc.mark_id=${mark_id}"
```

Comment:
```suggestion
    mark_id_config="-c spark.hadoop.lakefs.gc.mark_id=${mark_id} "
```
Otherwise it breaks when we add the third option.

Or use
```bash
function join_by { local IFS="$1"; shift; echo "$*"; }
```
from this [great StackOverflow answer](https://stackoverflow.com/a/17841619/192263).

Reply:
> Otherwise it breaks when we add the third option.

This is currently pretty static, and hopefully, we won't add more configurations (or code) to this test. But sure.

> `function join_by { local IFS="$1"; shift; echo "$*"; }`

Not in this lifetime